tests/generated/selfref.pb.zig

```diff
@@ -0,0 +1,21 @@
+// Code generated by protoc-gen-zig
+///! package selfref
+const std = @import("std");
+const Allocator = std.mem.Allocator;
+const ArrayList = std.ArrayList;
+
+const protobuf = @import("protobuf");
+const ManagedString = protobuf.ManagedString;
+const fd = protobuf.fd;
+
+pub const SelfRefNode = struct {
+    version: i32 = 0,
+    node: ?SelfRefNode = null,
```

Comment:
I think we could make this much easier if we used a pointer for self referencial nodes. Or if we always used a pointer.


Although the new recommendation for protobuf implementations, is to use moving or copying getters and setters in addition to opaque structs. Enabling internal reordering and packing

Reply:
I feel like we would lose some quality of use if we were to use a pointer everywhere. SelfRef structures are not uncommon, but i feel like they're far from being a common use case. I'd like to keep the common use case as simple and straightforward as possible.

My goal in this branch is to identify cases where there is some self ref structures in the structure hierarchy when running the plugin, and only using ManagedStruct in these case.